### PR TITLE
add has-method

### PR DESCRIPTION
**Pull request by prolic**

### CODE REVIEW

**Comment by marc-mabe** [comment]
For `isValidName`: you can simply do `defined("MyEnum::NAME")` instead.

**Comment by prolic** [comment]
So should I remove isValidName? I like this method because it allows me to do this:

    $enum->isValidName('FOO');

without having the enum-class at hand.

**Comment by prolic** [comment]
$foo = 'BAR';

    MyEnum::isValid($bar)

VS

    defined(MyEnum::class . '::' . $bar)

I would prefere the first version (with method), it's less code to write and more explicit, imho. But you decide @marc-mabe.

**Comment by prolic** [comment]
Damn scrutinizer won't succeed

**Comment by marc-mabe** [comment]
I don't see a real need for `isValidName()`
For `isValidOrdinal()`: You can simply check: `$ord < count(MyEnum::getConstants())`

For `isValid()` it makes sense because there is no simple and fast way to check (I only don't like the name).
What about:
```
final public static function has($enumerator) : bool

MyEnum::has(MyEnum::ONE); // true
MyEnum::has(MyEnum::ONE()); // true
MyEnum::has(AnotherEnum::ONE()); // false
MyEnum::has('unknown'); // false
```

**Comment by prolic** [comment]
changed as requested

File: src/Enum.php [hunk 1]
```diff
@@ -268,6 +268,22 @@ public function __toString()
...
+    final public static function has($value)
+    {
+        if ($value instanceof static && get_class($value) === get_called_class()) {
+            return true;
+        }
+
+        $class     = get_called_class();
+        $constants = self::detectConstants($class);
+        $name      = array_search($value, $constants, true);
+        return $name !== false;
```
Comment (marc-mabe) [inline src/Enum.php]: `return in_array($value, $constants, true);`

File: tests/MabeEnumTest/EnumTest.php [hunk 2]
```diff
@@ -245,4 +245,13 @@ public function testNotUnserializable()
         $this->setExpectedException('LogicException');
         unserialize("O:32:\"MabeEnumTest\TestAsset\EnumBasic\":0:{}");
     }
+
+    public function testHas()
+    {
+        $enum = EnumBasic::ONE();
+
+        $this->assertFalse($enum->isValid('invalid'));
+        $this->assertTrue($enum->isValid(1));
```
Comment (marc-mabe) [inline tests/MabeEnumTest/EnumTest.php]: please use `$this->assertTrue($enum->isValid(EnumBasic::ONE));`

**Comment by prolic** [comment]
done

**Comment by marc-mabe** [comment]
Please can you also address my small code comments and rebase all your commits together.
Thanks!

**Comment by prolic** [comment]
I hope this is okay this time ;)

**Pull request merged**